File: gpio_web.php
<?php

if(isset($_GET['pin'])){
  $pin = $_GET['pin'];
}
if(isset($_GET['state'])){
  $state = $_GET['state'];
}
if(isset($_GET['action'])){
  $action = $_GET['action'];
}

$pins = array("03"=>"08","05"=>"09","07"=>"07","08"=>"15","10"=>"16","11"=>"0","12"=>"01","13"=>"02","14"=>"","15"=>"03","16"=>"04","18"=>"05","19"=>"12","21"=>"13","22"=>"06","23"=>"14","24"=>"10","26"=>"11");
$pin2do = $pins[$pin];

if($action == 'state'){
  $tmp = exec("gpio read $pin2do");
  #echo "The current state of pin $pin($pin2do) is $tmp";
  $data = array("action"=>"state","header"=>$pin,"wiringpi_pin"=>$pin2do,"result"=>$tmp);
  $json = json_encode($data);
  echo $json;
}
elseif($action == 'write'){
if($state === '1'){
  $state = 1;
}

elseif($state === '0'){
  $state = 0;
}
else{
http_response_code(406);
echo"
<h1>Invalid state</h1>
Valid states are 1 and 0
";
exit();
}

$tmp = exec("gpio mode $pin2do out | gpio write $pin2do $state");
$tmp = exec("gpio read $pin2do");
#echo "Set pin $pin ($pin2do) to state $state";
$data = array("action"=>"write","header"=>$pin,"wiringpi_pin"=>$pin2do,"result"=>$tmp);
$json = json_encode($data);
echo $json;
}

elseif($action == 'read'){
  $tmp = exec("gpio mode $pin2do in | gpio read $pin2do");
  #echo "The current read of pin $pin($pin2do) is $tmp";
  $data = array("action"=>"read","header"=>$pin,"wiringpi_pin"=>$pin2do,"result"=>$tmp);
  $json = json_encode($data);
  echo $json;
}
else{
http_response_code(406);
echo"
<h1>Invalid action</h1>
Valid actions are:<br>
-Read (Reads the input of a pin)<br>
-Write (Writes a value to a pin)<br>
-State (Gets the state of a pin)<br>
";

}
?>

File: gpio_web.py
import flask
from flask import Flask
from flask import request
import RPi.GPIO as GPIO
app = Flask(__name__)
GPIO.setmode(GPIO.BOARD)


@app.route('/gpio_web', methods=['GET'])
def gpio_web():
  action = request.args['action']
  pin = int(request.args['pin'])
  state = request.args['state']
  print('Action: '+action)
  print('Pin: '+str(pin))
  print('State: '+str(state))
  if not (state == '1' or state == '0' or state == 'pullup' or state == 'pulldown'):
     return 'Invalid state '+str(state)
  if action == 'write':
     GPIO.setup(pin, GPIO.OUT)
     res = GPIO.output(pin, int(state)) 
     data = {'action':'write','pin':pin,'state':state,'result':res}
     data = flask.jsonify(data)
     return data
  if action == 'read':
     if state == 'pullup':
        GPIO.setup(pin, GPIO.IN, pull_up_down=GPIO.PUD_UP)
     if state == 'pulldown':
        GPIO.setup(pin, GPIO.IN, pull_up_down=GPIO.PUD_DOWN)
     else:
        GPIO.setup(pin, GPIO.IN)
     res = GPIO.input(pin)
     data = {'action':'read','pin':pin,'state':res,'result':res}
     data = flask.jsonify(data)
     return data
  if action == 'state':
     res = GPIO.input(pin)
     data = {'action':'state','pin':pin,'state':res,'result':res}
  else:
     return 'Invalid action'


app.run(debug=True,host='192.168.1.5',port=5000)


